xio-core/src/main/java/com/xjeffrose/xio/config/DynamicRouteConfigsFactory.java

```diff
@@ -0,0 +1,48 @@
+package com.xjeffrose.xio.config;
+
+import com.google.gson.Gson;
+import java.util.ArrayList;
+
+/**
+ * This class is used to generate a List<DynamicRouteConfig> based in a JSON string input matching
+ * the format described in /test/resources/route_parameters.json
+ */
+public class DynamicRouteConfigsFactory {
+
+  /**
+   * This is a factory method used to invoke the build operation given a JSON input string The
+   * output of this file is the raw material used to build ProxyRoutes dynamically from JSON
+   */
+  public static ArrayList<DynamicRouteConfig> build(String string) {
+    Gson gson = new Gson();
+    DynamicRouteEntry[] dynamicRouteEntries = gson.fromJson(string, DynamicRouteEntry[].class);
```

Comment:
i'd prefer the use of a `List<DynamicRouteEntry>` over an array - works with both Gson and Jackson

Reply:
yeah i can swap that